File: Programacao_Concorrente/java/src/imd/ufrn/br/bilheteria/Participant.java
package imd.ufrn.br.bilheteria;

public class Participant extends Thread {
	Ticket myticket;
	private int myseat;
	
	public Participant(Ticket ticket) {
		super();
		myticket = ticket;
		myseat = -1;
	}
	
	public void run() {
		myseat = myticket.getTicket();
		
		if(myseat == -1) {
			System.out.println(Thread.currentThread().getName() + ": Não há mais ingressos");
		} else {
			System.out.println(Thread.currentThread().getName() + ": Comprou assento. Restam" + myticket.getAvailableTickets() + " ingressos.");
		}
	}
}

File: Programacao_Concorrente/c#/Bilheteria/src/Participant.cs
namespace Bilheteria;

using System.Threading;

public class Participant(Ticket ticket)
{
    private readonly Ticket my_ticket = ticket;
    private int my_seat = -1;

    public void Run(){
        my_seat = my_ticket.GetTicket();

        if(my_seat == -1){
            Console.WriteLine(Thread.CurrentThread.Name + ": Não há mais ingressos.");
        } else {
            Console.WriteLine(Thread.CurrentThread.Name + ": Comprou assento. Restam " + my_ticket.GetAvailableTickets() + " ingressos.");
        }
    }
}